tables/akk.tbl

```diff
@@ -0,0 +1,958 @@
+
+# liblouis: Akkadian
+#
+# ------------
+#-index-name: Akkadian uncontracted
+#-display-name: Akkadian uncontracted
+#
+#+language:akk
+#+type:literary
+#+contraction:no
+#+system:akk-oracc
+
+#-license: lgpl-2.1
+# This file is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+
+# This file is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+
+# You should have received a copy of the GNU Lesser General Public
+# License along with this file; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+# liblouis  comes with ABSOLUTELY NO WARRANTY.
+
+
+#for more information on the Akkadian language, please go to:
+#https://oracc.museum.upenn.edu/dcclt/signlists/signlists/
+#The standard for Akkadian has been set by the academic community represented by ORACC. The braille code for Akkadian follows the standard set by ORACC. The braille code for Akkadian is represented in braille as the name for the sign in Akkadian. Thus, if the Akkadian sign is a "Lum," then the braille code for this sign would be lum.
+
+#This braille code was developed by a group of both blind and sighted scholars in an academic setting.
+#-maintainers: Paul Geoghegan <contact@envisionly.tech> and Matityhau Yeshurun <yeatersink@gmail.com>
+letter 𒀀 1
+letter 𒀁 1-16-1
+letter 𒀂 1-16-12-1-145
+letter 𒀃 1-16-1245-1-1345-0-2345-15-1345-136
+letter 𒀄 1-16-125-1
+letter 𒀅 1-16-24-1245-24
+letter 𒀆 1-16-123-1-1245-345-0-1245-136-1345-136
```

Comment:
I know @egli suggested that you don't use the `always` opcode like you did initially, but instead use character definition opcodes such as `letter`. However I'm not sure it was an improvement to replace all instances of `always` with `letter`. Personally I don't have anything against the `always` opcode. It's just the most generic opcode in Liblouis, and can be used when no other opcodes apply. I'm not sure if `letter` applies better here. Liblouis uses letter information to determine word boundaries.

Now, the rules in this table map from Cuneiform signs to the names of these signs. I don't know what constitutes a word in Cuneiform Akkadian, so I can't assess whether letter is appropriate or not. I can however observe that some dot patterns contain a "0", when the name of the sign contains a space. For example, 𒀆 ("A times lagar gunu") becomes 1-16-123-1-1245-345-0-1245-136-1345-136. A letter that is mapped to a dot pattern with a "0" in it, is fishy.

One thing that definitely looks wrong is the rule `letter \s 0`, which maps the space character to a blank. A space character is normally always in the `space` character class, using `letter` for a space is weird.

Reply:
I'm still seeing `letter` definitions with "0" in the dot pattern. I'm not confident this is the best choice. `always` would seem better.

Have you also considered doing the braille transcription in two phases? A first pass that does the translation from Cuneiform signs to English text (this would use the `correct` opcode), and a second pass that does the braille transcription using e.g. the UEB code.

Reply:
@bertfrees I thought you just wanted the ones that were defined as multiple unicode characters to be always.
We haven't considdered that at least not yet but that is escentially how we generate the braille for the characters.

Reply:
The ones that are defined as multiple unicode characters can't be `letter` because that is an error. Using `letter` for dot patterns with a "0" in it is not an error, just something that seems weird. In the end I don't really care. If the table does what it needs to do, it is fine. I'm just raising the concern that this might not be the right choice and might result in issues later on. But we can also consider that a problem for later.

I think my idea of using correct rules to transform Cuneiform signs to English could be interesting. I think it's something to consider for a next version.